### PR TITLE
Fix #1378

### LINKED ISSUE #1378: Configured `MaxMessageSize` is not honoured
### Describe the bug

Despite configuring ConnectionFactory.MaxMessageSize, this limit is not applied and instead the .Net client uses the default limit set in ConnectionFactory.DefaultMaxMessageSize (=134217728).

This does not happen in 6.x versions. The reason is because when the ConnectionFactory creates `AmqpTcpEndpoint`, it uses a constructor which does not take in the parameter `maxMessageSize` hence it defaults it to ConnectionFactory.DefaultMaxMessageSize. In 6.x, the default value was 0, i.e. unlimited. 

### Reproduction steps

1. Configure ConnectionFactory.DefaultMaxMessageSize to 500Mb
2. Send messages of 500Mb to a queue
3. Consume messages from that queue
4. The consumer fails with every single message because all of them are greater than 134217728
...

### Expected behavior

All messages should be consumed because they are equal to the configured `MaxMessageSize`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MarcialRosales**
- Fix issue #1378
- Updated `ConnectionFactory.DefaultMaxMessageSize` to today's RabbitMQ max message size. 
- Updated the APIApproval's `verified` file because of the MaxMessageSize changed in `ConnectionFactory.cs`. 
> Note: I replaced the `received` file with the `verified` one. According to the [docs](https://github.com/VerifyTests/Verify#initial-verification), that seems to be the correct thing to do.
- Bumped up the time to wait to receive a message in one of the test cases that ensure we cannot modify the message's payload after we sent it. In GH workflow, the test case always passed. But when running it from Concourse, things take longer to run and therefore 2sec was not enough.  

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [X] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

**Review by michaelklishin — APPROVED** [review]

File: projects/RabbitMQ.Client/client/api/ConnectionFactory.cs [hunk 1]
```diff
@@ -109,7 +109,7 @@ public sealed class ConnectionFactory : ConnectionFactoryBase, IConnectionFactor
         /// Corresponds to the <code>rabbit.max_message_size</code> setting.
         /// Note: the default is 0 which means "unlimited".
         /// </summary>
-        public const uint DefaultMaxMessageSize = 134217728;
+        public const uint DefaultMaxMessageSize = 536870912;
```
Comment (dank100) [inline projects/RabbitMQ.Client/client/api/ConnectionFactory.cs:112]: The comments state the the default value is 0, i.e. unlimited. Should we enforce this value?

File: projects/RabbitMQ.Client/client/api/AmqpTcpEndpoint.cs [hunk 2]
```diff
@@ -76,7 +76,7 @@ public AmqpTcpEndpoint(string hostName, int portOrMinusOne, SslOption ssl, uint
             HostName = hostName;
             _port = portOrMinusOne;
             Ssl = ssl;
-            _maxMessageSize = maxMessageSize;
+            _maxMessageSize = Math.Min(maxMessageSize, ConnectionFactory.DefaultMaxMessageSize);
```
Comment (dank100) [inline projects/RabbitMQ.Client/client/api/AmqpTcpEndpoint.cs:79]: Related to the comment below. 
This change is not necessary - If we continue to use 0 as the default value, i.e. unlimited, based on the comment.

File: projects/Unit/TestConnectionFactory.cs [hunk 3]
```diff
@@ -269,5 +269,51 @@ public void TestCreateConnectioUsesValidEndpointWhenMultipleSupplied()
             var ep = new AmqpTcpEndpoint("localhost");
             using (IConnection conn = cf.CreateConnection(new List<AmqpTcpEndpoint> { invalidEp, ep })) { };
         }
+
+        [Fact]
+        public void TestCreateConnectionUsesConfiguredMaxMessageSize()
```
Comment (dank100) [inline projects/Unit/TestConnectionFactory.cs:274]: Great tests, this is exactly what is needed to enforce this variable :)

File: projects/RabbitMQ.Client/client/api/ConnectionFactory.cs (hunk 1 above)
Comment (michaelklishin) [reply projects/RabbitMQ.Client/client/api/ConnectionFactory.cs:112]: Yes, we should use a limited default as of 6.6.0. It's high enough to likely not affect many.

Comment (lukebakken) [reply projects/RabbitMQ.Client/client/api/ConnectionFactory.cs:112]: The default max message size for RabbitMQ `main` (i.e. 3.13) is the following...
```
C:\Users\lbakken\development\rabbitmq\rabbitmq-server [rabbitmq-server-9258 ≡]
> git grep max_message_size
deps/rabbit/BUILD.bazel:138:        {max_message_size, 134217728},
deps/rabbit/Makefile:120:           {max_message_size, 134217728},
deps/rabbit/priv/schema/rabbit.schema:927:{mapping, "max_message_size", "rabbit.max_message_size"
```

...same with `3.12.x`
```
C:\Users\lbakken\development\rabbitmq\rabbitmq-server_v3.12.x [v3.12.x ≡]
> git grep max_message_size
deps/rabbit/BUILD.bazel:135:        {max_message_size, 134217728},
deps/rabbit/Makefile:118:           {max_message_size, 134217728},
deps/rabbit/priv/schema/rabbit.schema:927:{mapping, "max_message_size", "rabbit.max_message_size",
```

@MarcialRosales @michaelklishin why did we decide to use 512MiB in this code?

**Pull request merged**